AsyncDisplayKit/Details/ASRangeController.mm

```diff
@@ -129,11 +132,41 @@ - (void)addNodeToWorkingRange:(ASCellNode *)node
 - (void)moveNode:(ASCellNode *)node toView:(UIView *)view
 {
   ASDisplayNodeAssertMainThread();
-  ASDisplayNodeAssert(node && view, @"invalid argument, did you mean -removeNodeFromWorkingRange:?");
+  ASDisplayNodeAssert(node && view, @"invalid argument, did you mean -removeNodeFromRenderRange:?");
 
   [view addSubview:node.view];
 }
 
+- (void)addNode:(ASCellNode *)node toRange:(ASLayoutRange)range
+{
+  switch (range) {
+    case ASLayoutRangeRender:
+      [self addNodeToRenderRange:node];
+      break;
+    case ASLayoutRangePreload:
+      [node fetchExternalContent];
+      break;
+    default:
+      ASDisplayNodeAssert(NO, @"Cannot add a node to unsupported range %i",range);
+      break;
+  }
+}
+
+- (void)removeNode:(ASCellNode *)node fromRange:(ASLayoutRange)range
```

Comment:
nit: I would like to register an action and performSelector to generalize and simplify this function and below. Binding the selector to RangeType, and perform the selector.
Anyway, it is nit.